### ISSUE TITLE
[TreeView] does not use focus and focus visible states

### ISSUE THREAD

**Issue by amhci** [mui/mui-x#12561]
### Search keywords

focus visible

### Latest version

- [X] I have tested the latest version

### Steps to reproduce

Link to live example: Go to https://mui.com/x/react-tree-view/ and **Edit in StackBlitz**. (NOTE: I tried CodeSandbox too, but for this example at least, it gives an error, FYI.)

Steps:
1. Click on the first item "Data Grid". Note this node is now both selected (light blue background) and focus visible (light grey background).

### Current behavior

Clicking on nodes in TreeView visually indicate focus; there does not seem to be a differentiation between focus (not visible) and focus visible.

### Expected behavior

When a user clicks on a control, it should become focused, but not visibly focused. Only a change in state initiated by the keyboard should result in focus visible. Compare this against the Tabs component, for example, where this works properly. Click on a tab and it becomes focused, but you don't see it via focus visible. Hit an arrow key and now watch the new tab become focus visible because this was initiated via keyboard.

### Context

N/A

### Order ID or Support key 💳 (optional)

77929

**Comment by amhci** [comment]
IMO this problem is not easy to see or notice due to the current method of indicating both selected and focused via a change in background color, which is an accessibility and usability issue I reported separately in #12559. But once you indicate these states more clearly (see my example) it becomes very obvious!

**Comment by michelengelen** [comment]
Hey @amhci,
thanks for raising this as well. As with the other issue I have added this to our board. 👍🏼
